crates/cxx-qt-gen/src/syntax/types.rs

```diff
@@ -85,26 +85,26 @@ fn extract_qobject_from_mut_pin(ty: &TypePath) -> Result<(Ident, Mut)> {
                 if mutability.is_none() {
                     return Err(Error::new_spanned(
                         reference,
-                        "Expected a mutable reference when using Pin<>!",
+                        "Expected a mutable reference when using `Pin<>`!",
                     ));
                 }
                 return Ok((ident, mutability.unwrap()));
             } else {
                 return Err(Error::new_spanned(
                     ty,
-                    "Non reference args in Pin are not allowed",
+                    "Non reference args in Pin are not allowed!",
                 ));
             }
         } else {
             // CODECOV_EXCLUDE_START
-            unreachable!("Pin must use angle brackets for generic args like Pin<&mut T>");
+            unreachable!("Pin must use angle brackets for generic args like Pin<&mut T>!");
```

Comment:
This isn't actually `unreachable!`
If I write this:
```rust
    unsafe extern "RustQt" {
        #[qinvokable]
        fn increment_number(self: Pin);
    }
```
The bridge macro will panic!

So this should return a spanned error instead.
Wonder how we missed this for so long :thinking: 
You can also add a corresponding test and remove the CODECOVE_EXCLUDE region.

Reply:
hmm I guess we didn't think of that case when we marked it unreachable, I think maybe it was erroring in a use of parse_quote somewhere so we assumed it would be caught before that block. I've fixed this with a new test